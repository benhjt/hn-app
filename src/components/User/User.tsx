import { format, fromUnixTime } from 'date-fns';
import { UserType } from '../../types';

const formatDate = (timestamp: number) => {
  const date = fromUnixTime(timestamp);
  return format(date, 'MMMM dd, yyyy');
};

const User = ({ user }: { user: UserType }) => (
  <table className="table-auto border-collapse max-w- w-full">
    <tbody>
      <tr>
        <td>User:</td>
        <td>{user.id}</td>
      </tr>
      <tr>
        <td>Created:</td>
        <td>{formatDate(user.created)}</td>
      </tr>
      <tr>
        <td>Karma:</td>
        <td>{user.karma}</td>
      </tr>
    </tbody>
  </table>
);

export default User;
