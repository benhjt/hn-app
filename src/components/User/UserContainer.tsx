import { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import User from './User';
import { UserType } from '../../types';

const UserContainer = () => {
  const { username } = useParams<'username'>();
  const [isLoading, setIsLoading] = useState(false);
  const [user, setUser] = useState<UserType>();

  useEffect(() => {
    const fetchData = async () => {
      setIsLoading(true);
      const response = await fetch(
        `https://hacker-news.firebaseio.com/v0/user/${username}.json`,
      );
      const userData = await response.json();
      setUser(userData);
      setIsLoading(false);
    };

    fetchData();
  }, [username]);

  return (
    <div>
      {isLoading ? <p>Loading...</p> : user ? <User user={user} /> : ''}
    </div>
  );
};

export default UserContainer;
