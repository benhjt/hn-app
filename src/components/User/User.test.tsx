import { render } from '@testing-library/react';

import User from './User';

const user = {
  id: 'joebloggs123',
  created: 1640995200,
  karma: 54,
};

describe('FeedItem', () => {
  it('renders correctly', () => {
    const { container } = render(<User user={user} />);
    expect(container).toMatchSnapshot();
  });
});
