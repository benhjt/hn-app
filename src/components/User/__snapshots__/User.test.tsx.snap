// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedItem renders correctly 1`] = `
<div>
  <table
    class="table-auto border-collapse max-w- w-full"
  >
    <tbody>
      <tr>
        <td>
          User:
        </td>
        <td>
          joebloggs123
        </td>
      </tr>
      <tr>
        <td>
          Created:
        </td>
        <td>
          January 01, 2022
        </td>
      </tr>
      <tr>
        <td>
          Karma:
        </td>
        <td>
          54
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
